File: src/main/java/com/protryon/jasm/stage1/constant/CPackage.java
package com.protryon.jasm.stage1.constant;

import com.protryon.jasm.stage1.CConstant;

import java.io.DataOutputStream;
import java.io.IOException;

public class CPackage extends CConstant<Integer> {
	public CPackage(int ref) {
		super(ref, 20);
	}

	@Override
	public void write(DataOutputStream out) throws IOException {
		out.writeShort(this.value);
	}
	
}

File: src/main/java/com/protryon/jasm/stage1/constant/CPackage.kt
package com.protryon.jasm.stage1.constant

import com.protryon.jasm.stage1.CConstant

import java.io.DataOutputStream
import java.io.IOException

class CPackage(ref: Int) : CConstant<Int>(ref, 20) {

    @Throws(IOException::class)
    public override fun write(out: DataOutputStream) {
        out.writeShort(this.value)
    }

}
